custom_model_runner/datarobot_drum/drum/root_predictors/predict_mixin.py

```diff
@@ -321,7 +321,8 @@ def do_predict_structured(self, logger=None):
         return self._do_predict_structured(logger=logger)
 
     def do_predict_unstructured(self, logger=None):
-        if self._target_type != TargetType.UNSTRUCTURED:
+        # LLMs can also serve as guard models where moderations library expects to interface via /predictUnstructured
+        if self._target_type not in {TargetType.UNSTRUCTURED, TargetType.TEXT_GENERATION}:
```

Comment:
nit: I'd use a tuple rather than a set.
```suggestion
        if self._target_type not in (TargetType.UNSTRUCTURED, TargetType.TEXT_GENERATION):
```

Reply:
Why @zohar-mizrahi , set's have faster lookup in Python? (In C/C++ I would agree that short arrays are actually faster to search through linearly than a complex datastructure like a set but in Python with all the layers of indirection and pointers I'm not sure if that holds true)? Honestly if we wanted the fastest thing it would probably be to unwind and just do `target_type != UNSTRUCTURED or target_type != TEXT_GENERATION` but who cares about performance in Python 😉 )